Liquidation/contracts/Liquidation.sol

```diff
@@ -0,0 +1,303 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.14;
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@acala-network/contracts/dex/IDEX.sol";
+import "@acala-network/contracts/oracle/IOracle.sol";
+
+contract Liquidation is Ownable, Pausable {
+    struct CollateralPreference {
+        bool swapWithUSD;
+        bool limitedSupply;
+        uint256 supply;
+        uint256 totalSupplied;
+        uint256 minDiscount;
+    }
+
+    event Liquidate(
+        address collateral,
+        address payable repayDest,
+        uint256 supply,
+        uint256 target
+    );
+    event CollateralPreferenceUpdated(
+        address collateral,
+        CollateralPreference preference
+    );
+    event OnCollateralTransfer(address collateral, uint256 amount);
+    event OnRepaymentRefund(address collateral, uint256 amount);
+
+    address private _evm;
+    address private _USD_;
+    address private _DEX_;
+    address private _ORACLE_;
```

Comment:
Could you update this to use the `ADDRESS` utility directly? I think it would be ok for the tutorial to use the `MandalaAddress.sol`

This is already done in the [AdvancedEscrow](https://evmdocs.acala.network/tutorials/hardhat-tutorials/advancedescrow-tutorial#smart-contract)

Reply:
Oh I did that, then thought that it will increase contract size. 
Will use ADDRESS like AdvancedEscrow.sol

Reply:
For the tutorials, it's ok to focus on showcasing our features instead of optimizing the smart contracts, so it's ok if they are a bit bigger.

Reply:
I have used ADDRESS values in constructor now to initialize